gcc 10 portability

Bonpournickel opened the pull request:
default is -fno-common. To  avoid multiple definitions you need to explicitly initialize conflicting variables.

Commit:
Modifications qui seront validées :
	modifié :         src/cgame/cg_local.h
	modifié :         src/client/cl_main.c
	modifié :         src/client/snd_openal.c
	modifié :         src/irc/irc_client.c
	modifié :         src/irc/irc_client.h
	modifié :         src/qcommon/net_chan.c
	modifié :         src/qcommon/qcommon.h
	modifié :         src/renderer2/tr_shader.h
	modifié :         src/server/server.h
	modifié :         src/server/sv_main.c

Commit:
Merge https://github.com/etlegacy/etlegacy

Commit:
modifié :         cmake/ETLInstallOmniBot.cmake

Bonpournickel committed:
modifié :         .gitmodules

Bonpournickel committed:
Modifications qui seront validées :
	modifié :         libs

 Changements du sous-module à valider :

 * libs 9b810f7d...b02165ad (2):
   > 	modifié :         openal/Alc/bformatdec.h

isRyven commented on .gitmodules:
```diff
@@ -1,5 +1,5 @@
 [submodule "libs"]
 	path = libs
-	url = git://github.com/etlegacy/etlegacy-libs.git
+	url = https://github.com/Bonpournickel/etlegacy-libs.git
```
is this a mistake?

rmarquis replied:
Instead of patching any project in `libs`, you  need to send your patch upstream (in this case, OpenAL). We'll not maintain any forked code downstream.

Bonpournickel replied:
Sorry my mistake.
To make etlegacy-libs openal compatible with gcc-10 openal/Alc/bformatdec.h needs to be patched. It was the only way i figure out to point to it. I cannot use easily openal 32bit on my host. It dragged too much dependencies with it.

rmarquis commented on cmake/ETLInstallOmniBot.cmake:
```diff
@@ -27,6 +27,6 @@ LEG_DOWNLOAD(
 install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${MODNAME}/omni-bot/"
-	DESTINATION "${INSTALL_DEFAULT_MODDIR}/${MODNAME}/omni-bot"
-	DIRECTORY_PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ WORLD_READ
+	DESTINATION "/var/lib/etlegacy/omni-bot"
```
This breaks our standalone shipped package.

Bonpournickel replied:
Its purpose was to ease the use of omni-bot with a linux dedicated server. /var is rw and /usr is read-only for the server dedicated user in my case. 
An other variable  as ${INSTALL_DEFAULT_MODDIR} would be helpful.

rmarquis replied:
I am not sure what you are trying to solve here.
The shipped omnibot works fine when it isn't moved outside of `INSTALL_DEFAULT_MODDIR`.
If you want to sue another package that installs OB globally in your system path (with pre-compiled packages, like the `etlegacy-omnibot ` package in Arch Linux AUR), you surely don't need ETL to erase these packages again.

Requesting another variable does not make sense to me.

rmarquis commented on src/server/sv_main.c:
```diff
@@ -84,7 +84,7 @@ cvar_t *sv_wwwFallbackURL; // URL to send to if an http/ftp fails or is refused
 cvar_t *sv_cheats;
 cvar_t *sv_packetloss;
-cvar_t *sv_packetdelay;
+//cvar_t *sv_packetdelay;
```
Why? What happened to that cvar?

Bonpournickel replied:
It is declared in src/qcommon/common.c line 131 and conflict with etlegacy src/server/sv_main.c if I remember well.

Bonpournickel committed:
Merge branch 'master' of https://github.com/etlegacy/etlegacy

Bonpournickel committed:
modifié :         .gitmodules

 Modifications qui ne seront pas validées :
	modifié :         libs (nouveaux commits)

 Sous-modules modifiés mais non mis à jour :

 * libs b02165ad...f597cb8b (1):
   >  Modifications qui seront validées : 	modifié :         openal/.travis.yml 	supprimé :        openal/Alc/ALc.c 	supprimé :        openal/Alc/ALu.c 	supprimé :        openal/Alc/alconfig.c 	supprimé :        openal/Alc/alconfig.h 	supprimé :        openal/Alc/alstring.h 	supprimé :        openal/Alc/ambdec.c 	supprimé :        openal/Alc/ambdec.h 	supprimé :        openal/Alc/backends/alsa.c 	supprimé :        openal/Alc/backends/base.c 	supprimé :        openal/Alc/backends/base.h 	supprimé :        openal/Alc/backends/coreaudio.c 	supprimé :        openal/Alc/backends/dsound.c 	supprimé :        openal/Alc/backends/jack.c 	supprimé :        openal/Alc/backends/loopback.c 	supprimé :        openal/Alc/backends/null.c 	supprimé :        openal/Alc/backends/opensl.c 	supprimé :        openal/Alc/backends/oss.c 	supprimé :        openal/Alc/backends/portaudio.c 	supprimé :        openal/Alc/backends/pulseaudio.c 	supprimé :        openal/Alc/backends/sdl2.c 	supprimé :        openal/Alc/backends/sndio.c 	supprimé :        openal/Alc/backends/solaris.c 	supprimé :        openal/Alc/backends/wasapi.c 	supprimé :        openal/Alc/backends/wave.c 	supprimé :        openal/Alc/backends/winmm.c 	supprimé :        openal/Alc/bformatdec.c 	supprimé :        openal/Alc/bformatdec.h 	supprimé :        openal/Alc/compat.h 	supprimé :        openal/Alc/converter.c 	supprimé :        openal/Alc/converter.h 	supprimé :        openal/Alc/effects/autowah.c 	supprimé :        openal/Alc/effects/chorus.c 	supprimé :        openal/Alc/effects/compressor.c 	supprimé :        openal/Alc/effects/dedicated.c 	supprimé :        openal/Alc/effects/distortion.c 	supprimé :        openal/Alc/effects/echo.c 	supprimé :        openal/Alc/effects/fshifter.c 	supprimé :        openal/Alc/effects/modulator.c 	supprimé :        openal/Alc/effects/null.c 	supprimé :        openal/Alc/effects/pshifter.c 	supprimé :        openal/Alc/effects/reverb.c 	supprimé :        openal/Alc/filters/defs.h 	supprimé :        openal/Alc/filters/nfc.c 	supprimé :        openal/Alc/filters/nfc.h 	supprimé :        openal/Alc/filters/splitter.c 	supprimé :        openal/Alc/filters/splitter.h 	supprimé :        openal/Alc/fpu_modes.h 	supprimé :        openal/Alc/helpers.c 	supprimé :        openal/Alc/hrtf.c 	supprimé :        openal/Alc/hrtf.h 	supprimé :        openal/Alc/logging.h 	supprimé :        openal/Alc/mastering.c 	supprimé :        openal/Alc/mastering.h 	supprimé :        openal/Alc/mixer/defs.h 	supprimé :        openal/Alc/mixer/hrtf_inc.c 	supprimé :        openal/Alc/mixer/mixer_c.c 	supprimé :        openal/Alc/mixer/mixer_neon.c 	supprimé :        openal/Alc/mixer/mixer_sse.c 	supprimé :        openal/Alc/mixer/mixer_sse2.c 	supprimé :        openal/Alc/mixer/mixer_sse41.c 	supprimé :        openal/Alc/mixvoice.c 	supprimé :        openal/Alc/panning.c 	supprimé :        openal/Alc/polymorphism.h 	supprimé :        openal/Alc/ringbuffer.c 	supprimé :        openal/Alc/ringbuffer.h 	supprimé :        openal/Alc/uhjfilter.c 	supprimé :        openal/Alc/vector.h 	nouveau fichier : openal/BSD-3Clause 	modifié :         openal/CMakeLists.txt 	modifié :         openal/ChangeLog 	supprimé :        openal/OpenAL32/Include/alAuxEffectSlot.h 	supprimé :        openal/OpenAL32/Include/alBuffer.h 	supprimé :        openal/OpenAL32/Include/alEffect.h 	supprimé :        openal/OpenAL32/Include/alError.h 	supprimé :        openal/OpenAL32/Include/alFilter.h 	supprimé :        openal/OpenAL32/Include/alListener.h 	supprimé :        openal/OpenAL32/Include/alMain.h 	supprimé :        openal/OpenAL32/Include/alSource.h 	supprimé :        openal/OpenAL32/Include/alu.h 	supprimé :        openal/OpenAL32/Include/sample_cvt.h 	supprimé :        openal/OpenAL32/alAuxEffectSlot.c 	supprimé :        openal/OpenAL32/alBuffer.c 	supprimé :        openal/OpenAL32/alEffect.c 	supprimé :        openal/OpenAL32/alFilter.c 	supprimé :        openal/OpenAL32/alListener.c 	supprimé :        openal/OpenAL32/alSource.c 	supprimé :        openal/OpenAL32/alState.c 	supprimé :        openal/OpenAL32/event.c 	supprimé :        openal/OpenAL32/sample_cvt.c 	nouveau fichier : openal/al/auxeffectslot.cpp 	nouveau fichier : openal/al/auxeffectslot.h 	nouveau fichier : openal/al/buffer.cpp 	nouveau fichier : openal/al/buffer.h 	nouveau fichier : openal/al/effect.cpp 	nouveau fichier : openal/al/effect.h 	renommé :         openal/OpenAL32/alError.c -> openal/al/error.cpp 	nouveau fichier : openal/al/event.cpp 	nouveau fichier : openal/al/event.h 	renommé :         openal/OpenAL32/alExtension.c -> openal/al/extension.cpp 	nouveau fichier : openal/al/filter.cpp 	nouveau fichier : openal/al/filter.h 	nouveau fichier : openal/al/listener.cpp 	nouveau fichier : openal/al/listener.h 	nouveau fichier : openal/al/source.cpp 	nouveau fichier : openal/al/source.h 	nouveau fichier : openal/al/state.cpp 	nouveau fichier : openal/alc/alc.cpp 	nouveau fichier : openal/alc/alcmain.h 	nouveau fichier : openal/alc/alconfig.cpp 	nouveau fichier : openal/alc/alconfig.h 	nouveau fichier : openal/alc/alcontext.h 	nouveau fichier : openal/alc/alu.cpp 	nouveau fichier : openal/alc/alu.h 	nouveau fichier : openal/alc/ambdec.cpp 	nouveau fichier : openal/alc/ambdec.h 	nouveau fichier : openal/alc/ambidefs.h 	nouveau fichier : openal/alc/backends/alsa.cpp 	nouveau fichier : openal/alc/backends/alsa.h 	nouveau fichier : openal/alc/backends/base.cpp 	nouveau fichier : openal/alc/backends/base.h 	nouveau fichier : openal/alc/backends/coreaudio.cpp 	nouveau fichier : openal/alc/backends/coreaudio.h 	nouveau fichier : openal/alc/backends/dsound.cpp 	nouveau fichier : openal/alc/backends/dsound.h 	nouveau fichier : openal/alc/backends/jack.cpp 	nouveau fichier : openal/alc/backends/jack.h 	nouveau fichier : openal/alc/backends/loopback.cpp 	nouveau fichier : openal/alc/backends/loopback.h 	nouveau fichier : openal/alc/backends/null.cpp 	nouveau fichier : openal/alc/backends/null.h 	nouveau fichier : openal/alc/backends/opensl.cpp 	nouveau fichier : openal/alc/backends/opensl.h 	nouveau fichier : openal/alc/backends/oss.cpp 	nouveau fichier : openal/alc/backends/oss.h 	nouveau fichier : openal/alc/backends/portaudio.cpp 	nouveau fichier : openal/alc/backends/portaudio.h 	nouveau fichier : openal/alc/backends/pulseaudio.cpp 	nouveau fichier : openal/alc/backends/pulseaudio.h 	renommé :         openal/Alc/backends/qsa.c -> openal/alc/backends/qsa.cpp 	nouveau fichier : openal/alc/backends/qsa.h 	nouveau fichier : openal/alc/backends/sdl2.cpp 	nouveau fichier : openal/alc/backends/sdl2.h 	nouveau fichier : openal/alc/backends/sndio.cpp 	nouveau fichier : openal/alc/backends/sndio.h 	nouveau fichier : openal/alc/backends/solaris.cpp 	nouveau fichier : openal/alc/backends/solaris.h 	nouveau fichier : openal/alc/backends/wasapi.cpp 	nouveau fichier : openal/alc/backends/wasapi.h 	nouveau fichier : openal/alc/backends/wave.cpp 	nouveau fichier : openal/alc/backends/wave.h 	nouveau fichier : openal/alc/backends/winmm.cpp 	nouveau fichier : openal/alc/backends/winmm.h 	nouveau fichier : openal/alc/bformatdec.cpp 	nouveau fichier : openal/alc/bformatdec.h 	renommé :         openal/Alc/bs2b.c -> openal/alc/bs2b.cpp 	renommé :         openal/OpenAL32/Include/bs2b.h -> openal/alc/bs2b.h 	nouveau fichier : openal/alc/compat.h 	nouveau fichier : openal/alc/converter.cpp 	nouveau fichier : openal/alc/converter.h 	renommé :         openal/Alc/cpu_caps.h -> openal/alc/cpu_caps.h 	nouveau fichier : openal/alc/devformat.h 	nouveau fichier : openal/alc/effects/autowah.cpp 	nouveau fichier : openal/alc/effects/base.h 	nouveau fichier : openal/alc/effects/chorus.cpp 	nouveau fichier : openal/alc/effects/compressor.cpp 	nouveau fichier : openal/alc/effects/dedicated.cpp 	nouveau fichier : openal/alc/effects/distortion.cpp 	nouveau fichier : openal/alc/effects/echo.cpp 	renommé :         openal/Alc/effects/equalizer.c -> openal/alc/effects/equalizer.cpp 	nouveau fichier : openal/alc/effects/fshifter.cpp 	nouveau fichier : openal/alc/effects/modulator.cpp 	nouveau fichier : openal/alc/effects/null.cpp 	nouveau fichier : openal/alc/effects/pshifter.cpp 	nouveau fichier : openal/alc/effects/reverb.cpp 	nouveau fichier : openal/alc/effects/vmorpher.cpp 	renommé :         openal/Alc/filters/filter.c -> openal/alc/filters/biquad.cpp 	nouveau fichier : openal/alc/filters/biquad.h 	nouveau fichier : openal/alc/filters/nfc.cpp 	nouveau fichier : openal/alc/filters/nfc.h 	nouveau fichier : openal/alc/filters/splitter.cpp 	nouveau fichier : openal/alc/filters/splitter.h 	nouveau fichier : openal/alc/fpu_modes.h 	nouveau fichier : openal/alc/helpers.cpp 	nouveau fichier : openal/alc/hrtf.cpp 	nouveau fichier : openal/alc/hrtf.h 	renommé :         openal/Alc/inprogext.h -> openal/alc/inprogext.h 	nouveau fichier : openal/alc/logging.h 	nouveau fichier : openal/alc/mastering.cpp 	nouveau fichier : openal/alc/mastering.h 	nouveau fichier : openal/alc/mixer/defs.h 	nouveau fichier : openal/alc/mixer/hrtfbase.h 	nouveau fichier : openal/alc/mixer/mixer_c.cpp 	nouveau fichier : openal/alc/mixer/mixer_neon.cpp 	nouveau fichier : openal/alc/mixer/mixer_sse.cpp 	nouveau fichier : openal/alc/mixer/mixer_sse2.cpp 	renommé :         openal/Alc/mixer/mixer_sse3.c -> openal/alc/mixer/mixer_sse3.cpp 	nouveau fichier : openal/alc/mixer/mixer_sse41.cpp 	nouveau fichier : openal/alc/panning.cpp 	nouveau fichier : openal/alc/ringbuffer.cpp 	nouveau fichier : openal/alc/ringbuffer.h 	nouveau fichier : openal/alc/uhjfilter.cpp 	renommé :         openal/Alc/uhjfilter.h -> openal/alc/uhjfilter.h 	nouveau fichier : openal/alc/uiddefs.cpp 	nouveau fichier : openal/alc/voice.cpp 	nouveau fichier : openal/alc/voice.h 	modifié :         openal/alsoftrc.sample 	modifié :         openal/appveyor.yml 	supprimé :        openal/cmake/CheckFileOffsetBits.c 	supprimé :        openal/cmake/CheckFileOffsetBits.cmake 	supprimé :        openal/cmake/CheckSharedFunctionExists.cmake 	nouveau fichier : openal/cmake/FindMySOFA.cmake 	nouveau fichier : openal/cmake/FindOpenSL.cmake 	modifié :         openal/cmake/FindWindowsSDK.cmake 	nouveau fichier : openal/common/albyte.h 	supprimé :        openal/common/alcomplex.c 	nouveau fichier : openal/common/alcomplex.cpp 	modifié :         openal/common/alcomplex.h 	nouveau fichier : openal/common/alexcpt.cpp 	nouveau fichier : openal/common/alexcpt.h 	nouveau fichier : openal/common/alfstream.cpp 	nouveau fichier : openal/common/alfstream.h 	supprimé :        openal/common/align.h 	supprimé :        openal/common/almalloc.c 	nouveau fichier : openal/common/almalloc.cpp 	modifié :         openal/common/almalloc.h 	nouveau fichier : openal/common/alnumeric.h 	nouveau fichier : openal/common/aloptional.h 	nouveau fichier : openal/common/alspan.h 	nouveau fichier : openal/common/alstring.cpp 	nouveau fichier : openal/common/alstring.h 	supprimé :        openal/common/atomic.c 	modifié :         openal/common/atomic.h 	supprimé :        openal/common/bool.h 	nouveau fichier : openal/common/dynload.cpp 	nouveau fichier : openal/common/dynload.h 	nouveau fichier : openal/common/endiantest.h 	nouveau fichier : openal/common/intrusive_ptr.h 	modifié :         openal/common/math_defs.h 	nouveau fichier : openal/common/opthelpers.h 	nouveau fichier : openal/common/polyphase_resampler.cpp 	nouveau fichier : openal/common/polyphase_resampler.h 	nouveau fichier : openal/common/pragmadefs.h 	supprimé :        openal/common/rwlock.c 	supprimé :        openal/common/rwlock.h 	supprimé :        openal/common/static_assert.h 	nouveau fichier : openal/common/strutils.cpp 	nouveau fichier : openal/common/strutils.h 	supprimé :        openal/common/threads.c 	nouveau fichier : openal/common/threads.cpp 	modifié :         openal/common/threads.h 	supprimé :        openal/common/uintmap.c 	supprimé :        openal/common/uintmap.h 	nouveau fichier : openal/common/vecmat.h 	nouveau fichier : openal/common/vector.h 	modifié :         openal/common/win_main_utf8.h 	modifié :         openal/config.h.in 	modifié :         openal/docs/ambisonics.txt 	modifié :         openal/docs/env-vars.txt 	modifié :         openal/examples/alffplay.cpp 	modifié :         openal/examples/alhrtf.c 	modifié :         openal/examples/allatency.c 	modifié :         openal/examples/alloopback.c 	modifié :         openal/examples/almultireverb.c 	modifié :         openal/examples/alplay.c 	modifié :         openal/examples/alrecord.c 	modifié :         openal/examples/alreverb.c 	modifié :         openal/examples/alstream.c 	modifié :         openal/examples/altonegen.c 	modifié :         openal/examples/common/alhelpers.c 	modifié :         openal/examples/common/alhelpers.h 	nouveau fichier : openal/hrtf/Default HRTF.mhr 	supprimé :        openal/hrtf/default-44100.mhr 	supprimé :        openal/hrtf/default-48000.mhr 	modifié :         openal/include/AL/alc.h 	modifié :         openal/include/AL/alext.h 	modifié :         openal/include/AL/efx.h 	modifié :         openal/native-tools/bsincgen.c 	nouveau fichier : openal/resources/openal32.rc 	nouveau fichier : openal/resources/resource.h 	nouveau fichier : openal/resources/router.rc 	nouveau fichier : openal/resources/soft_oal.rc 	renommé :         openal/router/al.c -> openal/router/al.cpp 	renommé :         openal/router/alc.c -> openal/router/alc.cpp 	renommé :         openal/router/router.c -> openal/router/router.cpp 	modifié :         openal/router/router.h 	modifié :         openal/utils/CIAIR.def 	modifié :         openal/utils/IRC_1005.def 	modifié :         openal/utils/MIT_KEMAR.def 	nouveau fichier : openal/utils/MIT_KEMAR_sofa.def 	nouveau fichier : openal/utils/SCUT_KEMAR.def 	modifié :         openal/utils/alsoft-config/CMakeLists.txt 	modifié :         openal/utils/alsoft-config/mainwindow.cpp 	modifié :         openal/utils/alsoft-config/mainwindow.h 	modifié :         openal/utils/alsoft-config/mainwindow.ui 	nouveau fichier : openal/utils/alsoft-config/verstr.cpp 	nouveau fichier : openal/utils/alsoft-config/verstr.h 	supprimé :        openal/utils/bsincgen.c 	supprimé :        openal/utils/makehrtf.c 	nouveau fichier : openal/utils/makemhr/loaddef.cpp 	nouveau fichier : openal/utils/makemhr/loaddef.h 	nouveau fichier : openal/utils/makemhr/loadsofa.cpp 	nouveau fichier : openal/utils/makemhr/loadsofa.h 	nouveau fichier : openal/utils/makemhr/makemhr.cpp 	nouveau fichier : openal/utils/makemhr/makemhr.h 	modifié :         openal/utils/openal-info.c 	nouveau fichier : openal/utils/sofa-info.cpp 	nouveau fichier : openal/utils/sofa-support.cpp 	nouveau fichier : openal/utils/sofa-support.h 	modifié :         openal/version.h.in

Bonpournickel committed:
modifié :         libs

 Changements du sous-module à valider :

 * libs b02165ad...f597cb8b (1):

Bonpournickel committed:
* libs f597cb8b...3b9021db (1):
   > update to freetype-2.10.2 patched

Bonpournickel committed:
* libs 3b9021db...eb8f7487 (1):
   > 	modifié :         CMakeLists.txt 	new freetype 2.10.2 and openal 1.20.1 release

Bonpournickel committed:
Merge branch 'master' of https://github.com/etlegacy/etlegacy

Bonpournickel committed:
modifié :         .gitmodules

Bonpournickel committed:
modifié :         CMakeLists.txt
	modifié :         src/renderer2/tr_shader.c
	modifié :         src/renderer2/tr_shader.h
	modifié :         src/renderer2/tr_shader_r1.c

Bonpournickel committed:
Merge branch 'master' of https://github.com/etlegacy/etlegacy

rmarquis commented:
> ETL 32bit is playable and doesn't crash any more when opengl2 is called,

That's great and all, but your PR actually breaks the build for _everyone else_. In particular:
* Upgrading OpenAL breaks mingw-w64 on windows
* Upgrading Freetype/OpenAL seems to break msvc on release mode (debug seems fine): See AppVeyor error `LIBCMT.lib(ftol2.obj) : error LNK2005: __ftol2_sse already defined in SDL2.lib(SDL_stdlib.obj)`
* Your "better Linux integration" breaks our automated Linux builds. Nobody knows what you're trying to achieve in the first place.

Some GCC 10 fixes (somewhat based on your changes, but not really) have been pushed in 210e3effd4bf21862da31afef18ea967d0e456bb. Renderer2 code has not been adjusted yet.

rmarquis commented:
Closing this PR: GCC 10 compatibility has been added in a somewhat cleaner way. OpenAL update to latest version is more problematic (due to mingw and the C to C++ rewrite), but as this isn't directly related to GCC this can be handled later.
